### PR TITLE
Implement `RecoverY` hint

### PR DESCRIPTION
**Pull request by Sh0g0-1758**
Fixes: #490

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 1]
```diff
@@ -861,3 +861,108 @@ func newIsZeroDivModHint() hinter.Hinter {
...
+			}
+
+			xBigInt := new(big.Int)
+			xFelt.BigInt(xBigInt)
+
+			ySquaredBigInt := new(big.Int).Set(xBigInt)
+			ySquaredBigInt.Mul(ySquaredBigInt, xBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
+			ySquaredBigInt.Mul(ySquaredBigInt, xBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
+			ySquaredBigInt.Add(ySquaredBigInt, xBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
+			ySquaredBigInt.Add(ySquaredBigInt, betaBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_ec.go:921]: Maybe is a good idea to add a comment at the beginning here with the function you are implementing here and maybe even the line of code (https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/python/math_utils.py#L238). It's easier for readability

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 2]
```diff
@@ -861,3 +861,108 @@ func newIsZeroDivModHint() hinter.Hinter {
...
+					}
+
+					result.SetBigInt(tempResult)
+				} else {
+					ySquaredString := ySquaredBigInt.String()
+					return fmt.Errorf("%s does not represent the x coordinate of a point on the curve", ySquaredString)
+				}
+
+				value = mem.MemoryValueFromFieldElement(result)
+			}
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_ec.go:949]: Is this logic the same it was used for isQuadResidue hint? If it is maybe it's worth it to have it in a separate method, and just call it here

Comment (cicr99) [reply pkg/hintrunner/zero/zerohint_ec.go:949]: You'll have to return whether it is a quad residue in order to return the error if not, I suppose

File: pkg/hintrunner/zero/hintcode.go [hunk 3]
```diff
@@ -97,6 +97,7 @@ ids.multiplicities = segments.gen_arg([len(positions_dict[k]) for k in output])`
 	isZeroNondetCode         string = "memory[ap] = to_felt_or_relocatable(x == 0)"
 	isZeroPackCode           string = "from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack\n\nx = pack(ids.x, PRIME) % SECP_P"
 	isZeroDivModCode         string = "from starkware.cairo.common.cairo_secp.secp_utils import SECP_P\nfrom starkware.python.math_utils import div_mod\n\nvalue = x_inv = div_mod(1, x, SECP_P)"
+	RecoverYCode             string = "from starkware.crypto.signature.signature import ALPHA, BETA, FIELD_PRIME\nfrom starkware.python.math_utils import recover_y\nids.p.x = ids.x\n# This raises an exception if `x` is not on the curve.\nids.p.y = recover_y(ids.x, ALPHA, BETA, FIELD_PRIME)"
```
[the remark this answers is not in the document]
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/hintcode.go]: done.

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 1 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_ec.go]: Sure thing. Added a comment.

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 2 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_ec.go:951]: Yes True. Will make a separate method and update both the hints with a call to it. In which file should this method be ? Or do we make it public (starting letter caps) ?

Comment (cicr99) [reply pkg/hintrunner/zero/zerohint_ec.go:951]: I believe `pkg/hintrunner/utils/math_utils.go` should be the place where to add this. You'll have to make it public then

Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_ec.go]: Done.

**Review by cicr99 — APPROVED** [review]

File: pkg/hintrunner/utils/math_utils.go [hunk 4]
```diff
@@ -129,10 +129,49 @@ func sign(n *big.Int) (int, big.Int) {
...
+func IsQuadResidue(x *fp.Element) bool {
+	// Implementation adapted from sympy implementation which can be found here :
+	// https://github.com/sympy/sympy/blob/d91b8ad6d36a59a879cc70e5f4b379da5fdd46ce/sympy/ntheory/residue_ntheory.py#L689
+	// We have omitted the prime as it will be CAIRO_PRIME
+
+	if x.IsZero() || x.IsOne() || x.Legendre() == 1 {
+		return true
+	} else {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (Sh0g0-1758) [reply pkg/hintrunner/utils/math_utils.go]: Sure. That's much cleaner.

**Pull request merged**